plumbing/format/idxfile/idxfile.go

```diff
@@ -215,7 +216,6 @@ func (idx *MemoryIndex) genOffsetHash() error {
 	}
 
 	idx.offsetHash = make(map[int64]plumbing.Hash, count)
-	idx.offsetHashIsFull = true
```

Comment:
It might be a good idea to assign to a local variable then at the end assign it to `idx` so that lock contention is minimized

Reply:
Ok, did that but noticed that performance got worse in BenchmarkFindOffset after doing this. Perhaps I've done something wrong though.

Reply:
Oh, maybe that's due to other issues, FindOffset doesn't call FindHash. Fixed.